tests/contrib/hooks/test_gcp_bigquery_dts_hook.py

```diff
@@ -0,0 +1,136 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+from copy import copy
+
+from google.protobuf.json_format import ParseDict
+from google.cloud.bigquery_datatransfer_v1 import DataTransferServiceClient
+from google.cloud.bigquery_datatransfer_v1.types import TransferConfig
+
+from tests.compat import mock
+from tests.contrib.utils.base_gcp_mock import mock_base_gcp_hook_no_default_project_id
+
+from airflow.version import version
+from airflow.contrib.hooks.gcp_bigquery_dts_hook import BiqQueryDataTransferServiceHook
+
+
+CREDENTIALS = "test-creds"
+PROJECT_ID = "id"
+
+PARAMS = {
+    "field_delimiter": ",",
+    "max_bad_records": "0",
+    "skip_leading_rows": "1",
+    "data_path_template": "bucket",
+    "destination_table_name_template": "name",
+    "file_format": "CSV",
+}
+
+TRANSFER_CONFIG = ParseDict(
+    {
+        "destination_dataset_id": "dataset",
+        "display_name": "GCS Test Config",
+        "data_source_id": "google_cloud_storage",
+        "params": PARAMS,
+    },
+    TransferConfig(),
+)
+
+TRANSFER_CONFIG_ID = "id1234"
+
+
+class BigQueryDataTransferHookTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        with mock.patch(
+            "airflow.contrib.hooks.gcp_bigquery_dts_hook.GoogleCloudBaseHook.__init__",
+            new=mock_base_gcp_hook_no_default_project_id,
+        ):
+            self.hook = BiqQueryDataTransferServiceHook(  # type: ignore
+                gcp_conn_id=None
+            )
+            self.hook._get_credentials = mock.MagicMock(  # type: ignore
+                return_value=CREDENTIALS
+            )
+
+    def test_version_information(self):
+        expected_version = "airflow_v" + version
+        self.assertEqual(expected_version, self.hook.client_info.client_library_version)
+
+    def test_disable_auto_scheduling(self):
+        expected = copy(TRANSFER_CONFIG)
```

Comment:
```suggestion
        expected = deepcopy(TRANSFER_CONFIG)
```
I think that it is worth changing it in other places also.